### PR TITLE
IRepositoriesClient.RepoCollaborators -> IRepositoriesClient.Collaborator

### LINKED ISSUE #1036: IRepositoriesClient.RepoCollaborators -> IRepositoriesClient.Collaborator
The prefix is unnecessary, and is a pluralized property that we shouldn't really have:

https://developer.github.com/v3/repos/collaborators/

```csharp
var collaborators = await github.Repository.RepoCollaborators.GetAll("octokit", "octokit.net");
```

We should mark this property as obsolete and introduce a new property which is singular:

```csharp
var collaborators = await github.Repository.Collaborator.GetAll("octokit", "octokit.net");
```

### PR DESCRIPTION
**Pull request by M-Zuber**
Fixes #1036 

Updated.

### CODE REVIEW

**Comment by shiftkey** [comment]
@M-Zuber I think you can make this change without renaming types - perhaps some extra changes are causing the build failures to creep in...

**Comment by M-Zuber** [comment]
Okay, so the wanted change is just the properties on the `RepositoriesClient`s?
I'll start over in a new branch, if that's okay. A little simpler for me

**Comment by shiftkey** [comment]
> Okay, so the wanted change is just the properties on the `RepositoriesClients`?

Correct. Let's keep it simple!

> I'll start over in a new branch, if that's okay. A little simpler for me

If you're fine with force-pushing this branch to blow away the current changes, I'm fine with that. Otherwise just open a new PR.

**Comment by M-Zuber** [comment]
Force push is fun :smile: 
*I'll be back*

**Comment by shiftkey** [comment]
> Force push is fun :smile:

File: Octokit.Reactive/Clients/ObservableRepositoriesClient.cs [hunk 1]
```diff
@@ -22,7 +22,10 @@ public ObservableRepositoriesClient(IGitHubClient client)
             CommitStatus = new ObservableCommitStatusClient(client);
             Hooks = new ObservableRepositoryHooksClient(client);
             Forks = new ObservableRepositoryForksClient(client);
+#pragma warning disable CS0618 // Type or member is obsolete
```
Comment (M-Zuber) [inline Octokit.Reactive/Clients/ObservableRepositoriesClient.cs]: This ensures that current consumers won't get null

**Comment by M-Zuber** [comment]
@shiftkey that gif more then accurately describes what I just went through


CI y u no pas???

File: Octokit/Clients/RepositoriesClient.cs [hunk 2]
```diff
@@ -25,7 +25,10 @@ public RepositoriesClient(IApiConnection apiConnection) : base(apiConnection)
             CommitStatus = new CommitStatusClient(apiConnection);
             Hooks = new RepositoryHooksClient(apiConnection);
             Forks = new RepositoryForksClient(apiConnection);
+#pragma warning disable CS0618 // Type or member is obsolete
```
Comment (M-Zuber) [inline Octokit/Clients/RepositoriesClient.cs]: See above

**Comment by M-Zuber** [comment]
BTW I noticed that there seems to be no tests for this property. Is this on purpose?

File: Octokit.Reactive/Clients/ObservableRepositoriesClient.cs (hunk 1 above)
Comment (shiftkey) [inline Octokit.Reactive/Clients/ObservableRepositoriesClient.cs]: :thumbsup:

**Comment by shiftkey** [comment]
> BTW I noticed that there seems to be no tests for this property. Is this on purpose?

No, likely just something that was overlooked. Tracking that in #1045

**Comment by shiftkey** [comment]
@M-Zuber thanks!

**Pull request merged**